src/test/java/com/spotify/dns/DnsSrvResolversIT.java

```diff
@@ -74,6 +75,16 @@ public void shouldFailForBadHostNames() throws Exception {
     }
   }
 
+  @Test
+  public void shouldSucceedCreatingRetainingDnsResolver() throws Exception {
+    try {
+      resolver = DnsSrvResolvers.newBuilder().retainingDataOnFailures(true).build();
+    }
+    catch (DnsException e) {
```

Comment:
No need to even catch this exception - just validate that construction is possible without throwing anything. That would have been enough to catch this particular mistake. So it'll be enough with something like:

```
public void shouldSucceedCreatingRetainingDnsResolver() throws Exception {
  // validate that construction of a retaining resolver is possible
  DnsSrvResolvers.newBuilder().retainingDataOnFailures(true).build();
}
```

If I'm right, that test would have failed before your fix?